go/vt/servenv/http.go

```diff
@@ -0,0 +1,39 @@
+package servenv
+
+import (
+	"errors"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"net/http/pprof"
+)
+
+var mux = http.NewServeMux()
+
+func HTTPHandle(pattern string, handler http.Handler) {
+	mux.Handle(pattern, handler)
+}
+
+func HTTPHandleFunc(pattern string, handler func(http.ResponseWriter, *http.Request)) {
+	mux.HandleFunc(pattern, handler)
+}
+
+func HTTPServe(l net.Listener) error {
+	err := http.Serve(l, mux)
+	if errors.Is(err, http.ErrServerClosed) || errors.Is(err, net.ErrClosed) {
+		return nil
+	}
+	return err
+}
+
+func HTTPTestServer() *httptest.Server {
+	return httptest.NewServer(mux)
+}
```

Comment:
suggestion (to avoid having test library imports in the main `servenv` package:

```go
// go/vt/servenv/internal/mux.go
var Mux = http.NewServeMux()

// go/vt/servenv/http.go
import "vitess.io/vitess/go/vt/servenv/internal/mux"

func HTTPHandleFunc(...) {
    mux.Mux.HandleFunc(...)
}

// go/vt/servenv/testutils
import (
     "net/http"

    "vitess.io/vitess/go/vt/servenv/internal/mux"
)

func HTTPTestServer() *httptest.Server {
    return httptest.NewServer(mux.Mux)
}
```

Reply:
@ajm188 my idea was here to explicitly avoid having to export the mux so that isn’t used anywhere directly. 

But that has this downside indeed. Not sure which case should win here?

Reply:
Ah wait, you’re using internal, that’s probably the right trick then indeed.

Reply:
yep, because it's only exported internally, only servenv gets access

Reply:
@ajm188 Done!